### DISCUSSION TITLE
[mini.ai] Incremental selection

### DISCUSSION THREAD

**Discussion by dpetka2001** [nvim-mini/mini.nvim#323] [Q&A] [answered]
Example:
```lua
return {
  {
    "neovim/nvim-lspconfig",
    opts = {
      diagnostics = {
        virtual_text = false,
        virtual_lines = true,
      },
    },
  },
  {
    "ErichDonGubler/lsp_lines.nvim",
    url = "https://git.sr.ht/~whynothugo/lsp_lines.nvim",
    opts = {},
    keys = {
      { "<leader>l", "<cmd>lua require('lsp_lines').toggle()<cr>", desc = "Toggle lsp_lines" },
    },
  },
}
```
How could I incrementally select the 2 inner `{}` objects? With the cursor in `neovim/nvim-lspconfig` if I do `va}` it selects the region spec of nvim-lspconfig. I would like to increment the selection to also include the region with lsp_lines spec. Is that possible?

**Comment by echasnovski** [comment, marked as answer]
> How could I incrementally select the 2 inner `{}` objects? With the cursor in `neovim/nvim-lspconfig` if I do `va}` it selects the region spec of nvim-lspconfig. I would like to increment the selection to also include the region with lsp_lines spec. Is that possible?

Any 'mini.ai' textobject supports that by repeating it. More detailed: any application of textobject searches it so that to contain current selection. When you press `v` is selects current character. Then pressing `a}` searches for "around `}`" which contains current selection (single character). Pressing `a}` again will search for `a}` which contains current selection, and so on. This can also be done with `[count]` in Visual and Operator-pending mode (try `v2a}` and `d2a}`).

TL;DR: try typing `a}` once again.

**Reply by dpetka2001** [reply]
Thank you for your explanation. However, doing either `v2a}` or `va}` followed by another `a}` also selects the brackets that belong to `return` along with the 2 selected specs that I want. Is this normal behavior?

**Reply by echasnovski** [reply]
> Thank you for your explanation. However, doing either `v2a}` or `va}` followed by another `a}` also selects the brackets that belong to `return` along with the 2 selected specs that I want. Is this normal behavior?

Yes, with default "cover_or_next" search method it is the one that should be selected, as it is the smallest region surrounded by *balanced* curly brackets and *covering* current selection. If you want to select two consecutive regions, both surrounded by the balanced `{}`, then it is not immediately possible with 'mini.ai', as the result will not be the smallest region.

In this particular case you can try to use `i{` (not opening bracket) which will select inside `{}` without edge whitespace. But this won't work if there are more than 2 entries.

Another approach might be to move cursor manually to the `{` at the start of next argument and press `%` which will move it until matching `}`.

**Reply by dpetka2001** [reply]
I see what you mean. I think I understand a little better now how it works. Yes `vi{` will only work if there are only 2 entries for my purpose, otherwise going to the next `{` and pressing `%` will do the trick for my specific case. Thank you very much again for your insight.
